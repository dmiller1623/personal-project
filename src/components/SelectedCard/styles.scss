.selected-card {
  display: flex;
  flex-direction: column;
  background-color: #E4E4E1;
  background-image: radial-gradient(at top center, rgba(255,255,255,0.03) 0%, rgba(0,0,0,0.03) 100%), linear-gradient(to top, rgba(255,255,255,0.1) 0%, rgba(143,152,157,0.60) 100%);
  background-blend-mode: normal, multiply; 
  height: 24rem;
  width: 100%;
  margin: 1rem;
  border-radius: 15px;
  // width: 95%;
  padding: .2rem .6rem;
}

.selected-card-info {
  width: 100%;
  display: flex;
  flex-direction: row;
  // color: white;
}

.selected-card-title {
  width: 100%;
  display: flex;
  justify-content: space-between;
}

.selected-card-description {
  overflow: scroll;
  padding: .4rem;
  width: 65%;
  display: flex;
}

.selected-card-links {
  align-items: center;
  width: 35%;
  flex-direction: column;
  display: flex;
  // justify-content: flex-end;
}

.minus-button {
  margin-right: 2.2rem;
  width: 2rem;
  height: 2rem;
}

.minus-button:hover {
  cursor: pointer;
}

 .searched-title {
  margin-left: 2rem;
  color: black;
  font-weight: bold;
}

p {
  color: black;
}

