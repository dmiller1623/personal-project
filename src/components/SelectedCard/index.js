import React from 'react'
import minusSvg from '../../images/minus-symbol.svg'
import './styles.css'
import ShowMoreText from 'react-show-more-text';
import YouTube from 'react-youtube';


const SelectedCard = (props) => {
  return (
    <article className='selected-card'>
      <div className='selected-card-title'>
        {/* <div className='selected-button-section'> */}
          <img src={minusSvg} className='minus-button' alt='delete button' onClick={() => props.deleteResources(props)}/>
        {/* </div> */}
        {/* <div className='selected-search-heading'> */}
          <h1 className='searched-title'>{props.Name}</h1>
        {/* </div> */}
      </div>
      <div className='selected-card-info'>
        <div className='selected-card-description'> 
          <ShowMoreText>{props.wTeaser}</ShowMoreText>
        </div>
        <div className='selected-card-links'>
          {props.yUrl &&
            <YouTube
              className='youtube-video'
              videoId={props.yUrl.slice(39, 50)}
              opts={{
                height: '300',
                width: '300',
                autoplay: 1
              }}
            />}
            <a href={props.wUrl}>Wikipedia Link</a>
        </div>
      </div> 
    </article>
  )
}

export default SelectedCard