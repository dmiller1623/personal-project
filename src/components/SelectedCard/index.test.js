import React from 'react';
import SelectedCard from '../SelectedCard';
import { shallow } from 'enzyme';


describe('SelectedCard', () => {
  let wrapper;
  it('should match the snapshot', () => {
    let yUrl = 'kscislkcn';
    wrapper = shallow(<SelectedCard  yUrl={yUrl}/>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should call add resources when the add button is clicked', () => {
    let deleteResources = jest.fn();
    let yUrl = 'kscislkcn';

    wrapper = shallow(<SelectedCard yUrl={yUrl} deleteResources={deleteResources}/>);

    wrapper.find('img').simulate('click');
    expect(deleteResources).toHaveBeenCalled();
  });

});