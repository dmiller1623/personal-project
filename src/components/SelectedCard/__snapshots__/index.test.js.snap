// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedCard should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SelectedCard
    yUrl="kscislkcn"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="selected-card-title"
>
          <img
                    alt="delete button"
                    className="minus-button"
                    onClick={[Function]}
                    src="minus-symbol.svg"
          />
          <h1
                    className="searched-title"
          />
</div>,
        <div
          className="selected-card-info"
>
          <div
                    className="selected-card-description"
          >
                    <ShowMoreText
                              anchorClass=""
                              less="Show less"
                              lines={3}
                              more="Show more"
                              onClick={undefined}
                    />
          </div>
          <div
                    className="selected-card-links"
          >
                    <YouTube
                              className="youtube-video"
                              containerClassName=""
                              id={null}
                              onEnd={[Function]}
                              onError={[Function]}
                              onPause={[Function]}
                              onPlay={[Function]}
                              onPlaybackQualityChange={[Function]}
                              onPlaybackRateChange={[Function]}
                              onReady={[Function]}
                              onStateChange={[Function]}
                              opts={
                                        Object {
                                                  "autoplay": 1,
                                                  "height": "300",
                                                  "width": "300",
                                                }
                              }
                              videoId=""
                    />
                    <a
                              href={undefined}
                    >
                              Wikipedia Link
                    </a>
          </div>
</div>,
      ],
      "className": "selected-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="delete button"
              className="minus-button"
              onClick={[Function]}
              src="minus-symbol.svg"
/>,
            <h1
              className="searched-title"
/>,
          ],
          "className": "selected-card-title",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "delete button",
              "className": "minus-button",
              "onClick": [Function],
              "src": "minus-symbol.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "searched-title",
            },
            "ref": null,
            "rendered": null,
            "type": "h1",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="selected-card-description"
>
              <ShowMoreText
                            anchorClass=""
                            less="Show less"
                            lines={3}
                            more="Show more"
                            onClick={undefined}
              />
</div>,
            <div
              className="selected-card-links"
>
              <YouTube
                            className="youtube-video"
                            containerClassName=""
                            id={null}
                            onEnd={[Function]}
                            onError={[Function]}
                            onPause={[Function]}
                            onPlay={[Function]}
                            onPlaybackQualityChange={[Function]}
                            onPlaybackRateChange={[Function]}
                            onReady={[Function]}
                            onStateChange={[Function]}
                            opts={
                                          Object {
                                                        "autoplay": 1,
                                                        "height": "300",
                                                        "width": "300",
                                                      }
                            }
                            videoId=""
              />
              <a
                            href={undefined}
              >
                            Wikipedia Link
              </a>
</div>,
          ],
          "className": "selected-card-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ShowMoreText
                anchorClass=""
                less="Show less"
                lines={3}
                more="Show more"
                onClick={undefined}
/>,
              "className": "selected-card-description",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "anchorClass": "",
                "children": undefined,
                "less": "Show less",
                "lines": 3,
                "more": "Show more",
                "onClick": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <YouTube
                  className="youtube-video"
                  containerClassName=""
                  id={null}
                  onEnd={[Function]}
                  onError={[Function]}
                  onPause={[Function]}
                  onPlay={[Function]}
                  onPlaybackQualityChange={[Function]}
                  onPlaybackRateChange={[Function]}
                  onReady={[Function]}
                  onStateChange={[Function]}
                  opts={
                                    Object {
                                                      "autoplay": 1,
                                                      "height": "300",
                                                      "width": "300",
                                                    }
                  }
                  videoId=""
/>,
                <a
                  href={undefined}
>
                  Wikipedia Link
</a>,
              ],
              "className": "selected-card-links",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "youtube-video",
                  "containerClassName": "",
                  "id": null,
                  "onEnd": [Function],
                  "onError": [Function],
                  "onPause": [Function],
                  "onPlay": [Function],
                  "onPlaybackQualityChange": [Function],
                  "onPlaybackRateChange": [Function],
                  "onReady": [Function],
                  "onStateChange": [Function],
                  "opts": Object {
                    "autoplay": 1,
                    "height": "300",
                    "width": "300",
                  },
                  "videoId": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Wikipedia Link",
                  "href": undefined,
                },
                "ref": null,
                "rendered": "Wikipedia Link",
                "type": "a",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="selected-card-title"
>
            <img
                        alt="delete button"
                        className="minus-button"
                        onClick={[Function]}
                        src="minus-symbol.svg"
            />
            <h1
                        className="searched-title"
            />
</div>,
          <div
            className="selected-card-info"
>
            <div
                        className="selected-card-description"
            >
                        <ShowMoreText
                                    anchorClass=""
                                    less="Show less"
                                    lines={3}
                                    more="Show more"
                                    onClick={undefined}
                        />
            </div>
            <div
                        className="selected-card-links"
            >
                        <YouTube
                                    className="youtube-video"
                                    containerClassName=""
                                    id={null}
                                    onEnd={[Function]}
                                    onError={[Function]}
                                    onPause={[Function]}
                                    onPlay={[Function]}
                                    onPlaybackQualityChange={[Function]}
                                    onPlaybackRateChange={[Function]}
                                    onReady={[Function]}
                                    onStateChange={[Function]}
                                    opts={
                                                Object {
                                                            "autoplay": 1,
                                                            "height": "300",
                                                            "width": "300",
                                                          }
                                    }
                                    videoId=""
                        />
                        <a
                                    href={undefined}
                        >
                                    Wikipedia Link
                        </a>
            </div>
</div>,
        ],
        "className": "selected-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="delete button"
                className="minus-button"
                onClick={[Function]}
                src="minus-symbol.svg"
/>,
              <h1
                className="searched-title"
/>,
            ],
            "className": "selected-card-title",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "delete button",
                "className": "minus-button",
                "onClick": [Function],
                "src": "minus-symbol.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "searched-title",
              },
              "ref": null,
              "rendered": null,
              "type": "h1",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="selected-card-description"
>
                <ShowMoreText
                                anchorClass=""
                                less="Show less"
                                lines={3}
                                more="Show more"
                                onClick={undefined}
                />
</div>,
              <div
                className="selected-card-links"
>
                <YouTube
                                className="youtube-video"
                                containerClassName=""
                                id={null}
                                onEnd={[Function]}
                                onError={[Function]}
                                onPause={[Function]}
                                onPlay={[Function]}
                                onPlaybackQualityChange={[Function]}
                                onPlaybackRateChange={[Function]}
                                onReady={[Function]}
                                onStateChange={[Function]}
                                opts={
                                                Object {
                                                                "autoplay": 1,
                                                                "height": "300",
                                                                "width": "300",
                                                              }
                                }
                                videoId=""
                />
                <a
                                href={undefined}
                >
                                Wikipedia Link
                </a>
</div>,
            ],
            "className": "selected-card-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <ShowMoreText
                  anchorClass=""
                  less="Show less"
                  lines={3}
                  more="Show more"
                  onClick={undefined}
/>,
                "className": "selected-card-description",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "anchorClass": "",
                  "children": undefined,
                  "less": "Show less",
                  "lines": 3,
                  "more": "Show more",
                  "onClick": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <YouTube
                    className="youtube-video"
                    containerClassName=""
                    id={null}
                    onEnd={[Function]}
                    onError={[Function]}
                    onPause={[Function]}
                    onPlay={[Function]}
                    onPlaybackQualityChange={[Function]}
                    onPlaybackRateChange={[Function]}
                    onReady={[Function]}
                    onStateChange={[Function]}
                    opts={
                                        Object {
                                                            "autoplay": 1,
                                                            "height": "300",
                                                            "width": "300",
                                                          }
                    }
                    videoId=""
/>,
                  <a
                    href={undefined}
>
                    Wikipedia Link
</a>,
                ],
                "className": "selected-card-links",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "className": "youtube-video",
                    "containerClassName": "",
                    "id": null,
                    "onEnd": [Function],
                    "onError": [Function],
                    "onPause": [Function],
                    "onPlay": [Function],
                    "onPlaybackQualityChange": [Function],
                    "onPlaybackRateChange": [Function],
                    "onReady": [Function],
                    "onStateChange": [Function],
                    "opts": Object {
                      "autoplay": 1,
                      "height": "300",
                      "width": "300",
                    },
                    "videoId": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Wikipedia Link",
                    "href": undefined,
                  },
                  "ref": null,
                  "rendered": "Wikipedia Link",
                  "type": "a",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
